### PR TITLE
Update tests to handle older phpunit versions

### PR DESCRIPTION
**Pull request by qsomazzi**
## Changelog

```markdown
### Fixed
- Update tests method used to be compatible with older phpunit versions
```

## Subject

This MR allow older versions of phpunit to be used to run the tests

### CODE REVIEW

File: Tests/Helper/FragmentHelperTest.php [hunk 1]
```diff
@@ -39,11 +39,12 @@ protected function setUp()
...
     }
 
+    /**
+     * @expectedException \RuntimeException
+     * @expectedExceptionMessage Cannot render Fragment of type `foo.bar`. Service not found.
+     */
     public function testRenderWithoutService()
     {
-        $this->expectException('\RuntimeException');
-        $this->expectExceptionMessage('Cannot render Fragment of type `foo.bar`. Service not found.');
```
Comment (greg0ire) [inline Tests/Helper/FragmentHelperTest.php:45]: Please use `setExpectedException` with multiple arguments instead of annotations.

**Review by greg0ire — CHANGES_REQUESTED** [review]

**Comment by greg0ire** [comment]
Why do you need to be able to use old phpunit versions BTW? The oldest required version of php is 5.6.

**Comment by qsomazzi** [comment]
Because in our current project, in the test process, we launch phpunit tests in all our vendors. And our version of phpunit is 4.8. If we update to 5.8, we go more than 1000 deprecations notices that mess with our build report. So to be consistant (with other sonata bundles also) we just choose to downgrade tests so they can run on older phpunit version properly.

**Comment by greg0ire** [comment]
I think this weakens the argument about running vendor tests from inside the project even more. If you can install it with Composer, it means it went through Travis, and has already been tested. So unless one of the dependencies is broken to the point that using a different version will break the tests, it can't fail. 

But also, when you open-source a library, constraints that are very specific to your project shouldn't be taken into account (the fact that you run tests from the project dir, the fact that you use an old version of phpunit).

It just makes it harder to write tests for no real benefit.

**Comment by greg0ire** [comment]
Also, what prevents you from running phpunit twice, once with each version?

**Comment by Wharenn** [comment]
@greg0ire I don't really see the problem about prefering using wider supported methods in unit tests when possible. This allows to run tests without any error or warning on both PHPUnit 4 and 5. 

The Sonata travis setup is for now using PHPUnit 4.x for PHP < 5.6 and PHPUnit 5.x for PHP 5.6 or newer:
https://github.com/sonata-project/SonataAdminBundle/blob/master/.travis/install_test.sh#L7

However running PHPUnit 5.x on bundles like the admin is triggering a lot of warning:
https://travis-ci.org/sonata-project/SonataAdminBundle/jobs/186102988

IMO this is not a safe behavior and until Sonata is not fully supporting PHPUnit 5, it should not drop support of PHPUnit 4.

**Comment by greg0ire** [comment]
> @greg0ire I don't really see the problem about prefering using wider supported methods in unit tests when possible. This allows to run tests without any error or warning on both PHPUnit 4 and 5.

The only problem is that it makes test harder to write, and that you can't use any new PHPUnit 5 results, which is a pity.

What is the point of splitting Sonata in several packages if we forbid us from reaping the few benefits this setup provides?

> IMO this is not a safe behavior and until Sonata is not fully supporting PHPUnit 5, it should not drop support of PHPUnit 4.

Sonata is moving at a slow pace, I don't think making it harder is the way to go.

And again, why don't you run two test suites? One for bundles that are still compatible with php < 5.6, one for the others? That's a solution that would work, although I think that running the vendor libraries is useless. I know you are trying to catch some edge cases, but out of curiosity, have you ever caught one with that? Frankly, I doubt it.

**Comment by greg0ire** [comment]
Also, what if two of the dependencies you are testing have conflicting versions of their test dependencies, what then? Do you see how this does not make sense?

**Comment by core23** [comment]
Do we still need a support for outdated PHPUnit versions?

**Comment by qsomazzi** [comment]
In my opinion that would be a yes. 
But as discuss with gregoire, he think not. 
On our side we disable sonata-project tests unit for our pipeline, so it doesn't matter to us anymore.

So if other bundles doesn't support older version, i can close this PR.

**Comment by jordisala1991** [comment]
This is needed but it needs to be done focusing on next version, like it was done on other bundles. Create a wrapper for the testcase, redefine createMock function and do there the BC layer

**Comment by jordisala1991** [comment]
Oh, but if this is for master, then we dont need it, because we are only supporting php 7

**Comment by greg0ire** [comment]
Not needed, and external projects shouldn't have any influence on the bundle. Being able to run the sonata tests from your project is not a feature we should provide IMO, because it holds us back in terms of dependencies. Closing this.

**Pull request closed without merging**